### PR TITLE
Miscellaneous page table cleanups

### PR DESCRIPTION
**Pull request by lrh2000**
Just minor and miscellaneous cleanups. Some were noticed while reviewing #2042. I didn't leave comments there because they are not related to the changes in #2042.

I squashed them into one commit because they're all minor issues, and most of the changes are non-functional, such as comment adjustments. Creating too many small commits is probably not worth it. Hope this won't make the review difficult, as the changes in this PR are still relatively minor.

### CODE REVIEW

File: ostd/src/mm/page_table/cursor/mod.rs [hunk 1]
```diff
@@ -287,22 +288,14 @@ impl<'rcu, C: PageTableConfig> Cursor<'rcu, C> {
...
+            let node_start = self.va.align_down(node_size);
             // If the address is within the current node, we can jump directly.
-            if cur_node_start <= va && va < cur_node_end {
-                self.va = va;
-                return Ok(());
-            }
-
-            // There is a corner case that the cursor is depleted, sitting at the start of the
-            // next node but the next node is not locked because the parent is not locked.
-            if self.va >= self.barrier_va.end && self.level == self.guard_level {
```
Comment (lrh2000) [inline ostd/src/mm/page_table/cursor/mod.rs:300]: I initially moved this out of the loop and later realized it was unnecessary. So I removed it. However, it might be worth thinking about it again to see if I'm missing something.

At least, it seems that all the unit and regression tests are happy.

Comment (junyang-zh) [reply ostd/src/mm/page_table/cursor/mod.rs:300]: It was introduced months ago, and survived a lot of reactors. Once upon a time, there were calls in aster-nix that moves the cursor to the end of a region, and did a `jump`. It seems that no code can trigger this corner case now.

But I think we should add a test for this case:

```rust
#[ktest]
fn jump_from_the_end() {
    let pt = set_up_page_table::<TestPtConfig>();
    let lock_range = 0..PAGE_SIZE * 512 * 2; // 4M, so lock at level 2.
    let map_va = lock_range.end - PAGE_SIZE;
    let map_pa_range = 0..PAGE_SIZE;
    let mut cursor = pt.cursor_mut(&lock_range).unwrap();
    cursor.jump(map_va).unwrap();
    cursor.map(pa_range).unwrap();
    // Now the cursor is at the end of the range with level 2.
    cursor.jump(lock_range.start).unwrap();
}
```

Comment (junyang-zh) [reply ostd/src/mm/page_table/cursor/mod.rs:300]: Of course, the code here is confusing, even if the corner case is considered. I would appreciate your work on fixing it.

Comment (lrh2000) [reply ostd/src/mm/page_table/cursor/mod.rs:300]: > But I think we should add a test for this case:

Thanks for writing the test case! I tried adding it and checking its behavior. It appears that it is still hitting an earlier branch:
https://github.com/asterinas/asterinas/blob/a2601b86d4ed5446de3a7a70a9e1be4bee61b58f/ostd/src/mm/page_table/cursor/mod.rs#L291-L297

The page table node at level 2 covers `lock_range.start`, so it returns. So I believe that the corner case seems to be already covered by the normal path?

Comment (lrh2000) [reply ostd/src/mm/page_table/cursor/mod.rs:300]: I enhanced your test to verify the only intended functional change in this PR:
```rust
        cursor.jump(virt_range.end - HUGE_PAGE_SIZE / 2).unwrap();
        assert_eq!(
            cursor.query().unwrap().0,
            virt_range.end - HUGE_PAGE_SIZE..virt_range.end
        );
```
i.e., `query()` should return the whole range of a huge page, even if the cursor is in the middle of the huge page.

See [the updated changes](https://github.com/asterinas/asterinas/compare/a2601b86d4ed5446de3a7a70a9e1be4bee61b58f..0575686a8f51c3ffef300c42aa531dd18ba62bcb) for details.

File: ostd/src/mm/page_table/node/entry.rs [hunk 2]
```diff
@@ -138,28 +137,25 @@ impl<'a, 'rcu, C: PageTableConfig> Entry<'a, 'rcu, C> {
...
         };
 
         *self.node.nr_children_mut() += 1;
 
-        // SAFETY: The page table won't be dropped before the RCU grace period
-        // ends, so it outlives `'rcu`.
-        let pt_ref = unsafe { PageTableNodeRef::borrow_paddr(paddr) };
-        // SAFETY: The node is locked and there are no other guards.
-        Some(unsafe { pt_ref.make_guard_unchecked(guard) })
+        Some(pt_ref.lock(guard))
```
Comment (junyang-zh) [inline ostd/src/mm/page_table/node/entry.rs]: There seems to be a semantics change. The new node is locked after its address has been written to the parent node. That means another thread can lock the new node and do some operations before we return. The problem is that if a cursor locks a range, unmaps a node, and then maps again to insert a new node, it may find that the newly inserted node contains something rather than nothing. This violates the mutual exclusiveness of the range lock.

**Review by junyang-zh — COMMENTED** [review]
Only one comment. Other changes look good~

File: ostd/src/mm/page_table/node/entry.rs (hunk 2 above)
Comment (lrh2000) [reply ostd/src/mm/page_table/node/entry.rs]: Good catch! I didn't notice that at all.

We should still create the guard before writing the address to the page table, as `split_if_mapped_huge` does. Thanks for pointing this out!

**Review by junyang-zh — APPROVED** [review]
LGTM! Thanks for contribution.
